libraries/mbed/targets/cmsis/TARGET_NORDIC/TARGET_MCU_NRF51822/system_nrf51.c

```diff
@@ -61,6 +61,20 @@ void SystemCoreClockUpdate(void)
 
 void SystemInit(void)
 {
+#if defined(TARGET_NRF_32MHZ_XTAL)
```

Comment:
This code wriging the value 0xFFFFFF00 to the UICR (User Information Configuration Register) at address 0x10001008, to make nRF51 works with 32MHz system clock. Physically, tiny footprint XTAL oscillate higher freq. To make BLE modules smaller, some modules are using 32MHz XTAL.
This register will be overwritten by SoftDevice. Each hex files built with mbed classic online compiler contain SoftDevice, so that, this code run once just after the hex file flashed onto nRF51.